integration-webcert/src/main/java/se/inera/intyg/minaintyg/integration/webcert/converter/data/value/IcfValueConverter.java

```diff
@@ -17,14 +18,23 @@ public CertificateDataValueType getType() {
 
   @Override
   public CertificateQuestionValue convertToValue(CertificateDataElement element) {
-    final var value = ((CertificateDataIcfValue) element.getValue()).getText();
-    // TODO: Handle icf-codes
-    return getText(value);
+    final var icfCodes = ((CertificateDataIcfValue) element.getValue()).getIcfCodes();
+    final var text = ((CertificateDataIcfValue) element.getValue()).getText();
+
+    if (icfCodes != null) {
```

Comment:
Icf codes is an ArrayList that either contains icf codes or not, so I think we need to add a check if its empty aswell!

Should be easy to write a test for it aswell!

![image](https://github.com/sklintyg/minaintyg/assets/57347249/d8527c15-d171-45ed-9a92-28f0c0a3b944)